app/frontend/errors/errorMessages.ts

```diff
@@ -79,9 +80,12 @@ const internalErrorMessages: {[key in InternalErrorReason]: (params?: any) => st
   [InternalErrorReason.CryptoProviderError]: ({message}) => `CryptoProviderError: ${message}`,
   [InternalErrorReason.NetworkError]: () =>
     'NetworkError: Request to our servers has failed. Please check your network connection and if the problem persists, contact us.',
-  [InternalErrorReason.ServerError]: () =>
-    'ServerError: Our servers are probably down. Please try again later and if the problem persists, contact us.',
-
+  [InternalErrorReason.ServerError]: () => {
```

Comment:
overriding the translation message based on some external context feels a bit sketchy - these are meant to be static/pure functions after all. I'd rather have a separate error reason for epoch boundary and create a helper function which would throw that error reason instead of the generic ServerError if the epoch boundary is detected. We can use that helper wherever we throw the server error (which should be in 2 places AFAIK) which would also be more transparent to whoever is inspecting the errors being thrown